### PR TITLE
eureka: remove character name on map

### PR DESCRIPTION
**Pull request by Jaehyuk-Lee**
remove these characters on map ffxiv_20200423_211302_148
party member's \<flag\> chat -> those party member's number also printed on map

one better Korean label name.

### CODE REVIEW

**Comment by Jaehyuk-Lee** [comment]
character's server(world) name is not removed in Korean server. I'm checking it.

**Comment by Jaehyuk-Lee** [comment]
`00|2020-04-23T21:40:13.0000000+09:00|000b|(CharacterName)(WorldName)|(Chat)|38f38beb0d62f99066aba265c263b36e`
Is it impossible to remove world name? It seems there's no delimiter between Character name and its World name.

**Comment by quisquous** [comment]
> `00|2020-04-23T21:40:13.0000000+09:00|000b|(CharacterName)(WorldName)|(Chat)|38f38beb0d62f99066aba265c263b36e` - from parser log
> Is it impossible to remove world name? It seems there's no delimiter between Character name and its World name.

Yeah, it's the same in English.  In English at least, all the names are stripped from the flag text, so it doesn't get shown.

**Comment by Jaehyuk-Lee** [comment]
I thought showing character name (who said) was intended.
Should character name not be shown on map?

**Comment by quisquous** [comment]
In English it gets too long, so I'd say no? Really the goal of flags on the map was that people post a ton of flags about where the train is, where rezes are needed, etc and I just wanted a way to show those easily.  Who is saying those things seems less important.

**Comment by Jaehyuk-Lee** [comment]
okay, I'll try removing character name after short trip I'm having right now, maybe the day after tomorrow?

**Comment by quisquous** [comment]
(oops sorry, wrong button)

Sounds good! We probably don't need the number replacement if the party name is missing, I think, so I'll wait for you to get back.  Have a nice trip!

**Comment by Jaehyuk-Lee** [comment]
is there any other variables that gets FFXIV parsing plugin's language? I think `this.options.Language` is not this. I think some codes are using this variable mixed with server determine thing and language determine thing.

In this code, I'm using `this.options.Language` for determining Korean server, but I should take a look for server determining.

File: ui/eureka/eureka.js [hunk 1]
```diff
@@ -2056,8 +2056,12 @@ class EurekaTracker {
       let gRegex = this.Translate(this.options.Regex);
       let gFlagRegex = gRegex['gFlagRegex'];
       let match = log.match(gFlagRegex);
-      if (match)
+      if (match) {
+        // remove character name on Korean server (if not /echo)
```
Comment (quisquous) [inline ui/eureka/eureka.js:2060]: Hmm, it is surprising to me that there is different logic for different languages.  Is it possible to make them all behave the same?

**Comment by quisquous** [comment]
> is there any other variables that gets FFXIV parsing plugin's language? I think `this.options.Language` is not this. I think some codes are using this variable mixed with server determine thing and language determine thing.

`this.options.Language` should be it.  That is what is sent from the C# code based on what your parsing language is.  Which code is using it wrong?

**Comment by Jaehyuk-Lee** [comment]
Oh, I thought different thing.
alertsLanguage and timelineLanguage are configurable to other languages, so I thought some mistakenly.

File: ui/eureka/eureka.js (hunk 1 above)
Comment (Jaehyuk-Lee) [reply ui/eureka/eureka.js:2060]: > Yeah, it's the same in English. In English at least, all the names are stripped from the flag text, so it doesn't get shown.

As you said before, it seems only Korean (or Chinese too) server is showing character name on map, so I've set only to work on Korean server.
This logic is removing string ahead of `:`, which always logged except /echo.
so, if this logic is also applied to English version, 
`some place name: <flag>` will be replaced to ` <flag>`.

Comment (quisquous) [reply ui/eureka/eureka.js:2060]: The English flag regex is:
`'gFlagRegex': Regexes.parse(/00:00..:(?:[^:]*:)?(.*)Eureka (?:Anemos|Pagos|Pyros|Hydatos) \( (\y{Float})\s*, (\y{Float}) \)(.*$)/),`

Yeah, I think you're correct that if somebody did `/say text: <flag>` then the `text` part would be removed.

Maybe `/00:00(?:38:|..:[^:]*:)(.*)Eureka ... etc etc` would remove the name from non-echoes but work for echo, and we could apply that to all languages?

Comment (Jaehyuk-Lee) [reply ui/eureka/eureka.js]: This is working!

**Comment by quisquous** [comment]
Thanks for the patch and for the testing!

**Pull request merged**